demo/src/Types.elm

```diff
@@ -54,6 +56,33 @@ to the program as `endpoint` (see above).
 type Msg
     = Endpoint
     | GotQuotes (Result Http.Error (List Quote))
+    | RandomNumber Int
+
+
+{-| Your custom interop type.
+
+Should enumerate the interop (JavaScript) functions which can be called.
+
+-}
+type Interop
+    = GetRandom Int
+
+
+interopEncode : Interop -> Encode.Value
+interopEncode interop =
+    case interop of
+        GetRandom upper ->
+            Encode.int upper
+
+
+interopDecoder : String -> Maybe (Decode.Decoder Msg)
+interopDecoder interopName =
+    case interopName of
+        "getRandom" ->
+            Just <| Decode.map RandomNumber Decode.int
```

Comment:
The value that gets returned from JavaScript needs to be decoded, and wrapped in an app `msg`